file.go

```diff
@@ -63,23 +63,12 @@ type dbHeader struct {
 	SqliteVersion int32    // SQLITE_VERSION_NUMBER
 }
 
-func Open(fname string) (*DbFile, error) {
-	f, err := os.Open(fname)
-	if err != nil {
-		return nil, err
-	}
-
-	defer func() {
-		if err != nil {
-			f.Close()
-		}
-	}()
-
+func OpenInterface(f ReadSeekCloser, size int64) (*DbFile, error) {
```

Comment:
I'd be inclined to call it 'OpenFrom'.

Reply:
Also, could you add some documentation about it?
(Especially the 'size' parameter)

Reply:
BTW: is 'size' needed? Couldn't it be inferred by seeking to the end of the file?

Reply:
One more thing: having the parameter implement io.Closer implies that we'll close the underlying handle when Close is called on sqlite.DbFile.

That would come a bit as a surprise for people used to stdlib's API.

I'd be inclined to have 'OpenFrom' take an io.ReadSeeker only.
And add an io.Closer field to sqlite.DbFile, left nil in 'OpenFrom' and properly initialized in 'Open'.
DbFile.Close would need to be adapted, of course.